### PR TITLE
PC-1926 manual geofence request docs android

### PR DESCRIPTION
**Pull request by radixdev**
# Pull Request/Issue Resolution

**Description of Change:**
docs for https://appboy.github.io/appboy-android-sdk/javadocs/com/appboy/Appboy.html#requestGeofences-double-double-

https://jira.braze.com/browse/PC-1926

### Is this change associated with a Braze feature/product release?
- [ ] Yes (__Insert Feature Release Date Here__)
- [ ] No

> If yes, please note the date of the feature release.

---
---

<details>
<summary>PR Checklist</summary>

- [ ] Check that all links work!
- [ ] Ensure you have completed [our CLA](https://www.braze.com/docs/cla/).
- [ ] Tag @KellieHawks as a reviewer when the your work is _done and ready to be reviewed for merge_.
- [ ] Tag others as Reviewers as necessary.
- [ ] If you have modified any links, be sure to add redirects to `config/nginx.conf.erb`.

</details>

<details>
<summary>Writing & Style Quick Tips</summary>

- Use the formatting shown on the [Styling Test Page](https://github.com/Appboy/braze-docs/blob/develop/_docs/_home/styling_test_page.md) to create unique templates and vehicles for your information.
- Use Action verbs!
- Capitalize proper nouns, unique Braze tools and features, custom Braze channels, and all Header titles.
- Redirect when necessary!
- Always speak to the Braze customer - refer to them as “you”, whether they are an Engineer/Developer, a Marketer, or an Admin.
- On Voice: You should aim to be a Friendly Professor. Imagine, if you will, the smartest, but also kindest professor you had. You know, the one who showed you your potential and tested you, but also went slowly in class so you could take great notes and learn some things? Don't be too peppy, but don't be formal. You’re not selling anything - you’re instructing and informing and enabling.

_Consult the [Docs Text Formatting Guide](https://github.com/Appboy/braze-docs/wiki/Special-Formatting) and the [Docs Writing Style Guide & Best Practices](https://github.com/Appboy/braze-docs/wiki/Writing-Style-Guide-&-Best-Practices) for more details._

</details>

<details>
<summary>ATTENTION: For PR Reviewers</summary>

- [ ] Read our [Reviewing a PR page](https://github.com/Appboy/braze-docs/wiki/Reviewing-a-PR) for more on our reviewing suggestions.
- [ ] Preview all changes in the linked Heroku environment (click `View deployment` button below, then "Docs". A `502` error just means you should refresh until you see the Docs Home page.
</details>

---
---

### CODE REVIEW

File: _docs/_developer_guide/platform_integration_guides/android/advanced_use_cases/locations_and_geofences.md [hunk 1]
```diff
@@ -104,12 +104,201 @@ This will cause the SDK to request geofences from Braze's servers and initialize
...
+          Log.i(TAG, "Required location permissions granted.");
+          Toast.makeText(context, "Required location permissions granted.", Toast.LENGTH_SHORT).show();
+          AppboyLocationService.requestInitialization(context);
+        } else {
+          Log.i(TAG, "Required location permissions NOT granted.");
+          Toast.makeText(context, "Required location permissions NOT granted.", Toast.LENGTH_SHORT).show();
+        }
+        break;
+      default:
+        break;
```
Comment (davidbielik) [inline _docs/_developer_guide/platform_integration_guides/android/advanced_use_cases/locations_and_geofences.md]: :-0

File: _docs/_developer_guide/platform_integration_guides/android/advanced_use_cases/locations_and_geofences.md [hunk 2]
```diff
@@ -104,12 +104,201 @@ This will cause the SDK to request geofences from Braze's servers and initialize
...
+
 ### Step 5: Enable Geofences on the Dashboard
 
 Android only allows up to 100 geofences to be stored for a given app. Braze's Locations product will use up to 20 of these geofence slots if available. To prevent accidental or unwanted disruption to other geofence-related functionality in your app, location geofences must be enabled for individual Apps on the Dashboard.
 
 For Braze's Locations product to work correctly, you should also ensure that your App is not using all available geofence spots.
 
+### Step 6: (Optional) Manually Request Geofence Updates
+
+By default, Braze automatically retrieves the device's location and requests geofences based on that collected location. However, you can manually provide a GPS coordinate that will be used to retrieve proximal Braze Geofences. To manually request Braze Geofences, you must disable automatic Braze Geofence request and provide a GPS coordinate for requests.
```
Comment (davidbielik) [inline _docs/_developer_guide/platform_integration_guides/android/advanced_use_cases/locations_and_geofences.md]: Would it make more sense to say:
> However, you can manually provide a GPS coordinate that will be used to retrieve proximal Braze Geofences **instead**.

File: _docs/_developer_guide/platform_integration_guides/android/advanced_use_cases/locations_and_geofences.md [hunk 3]
```diff
@@ -104,12 +104,201 @@ This will cause the SDK to request geofences from Braze's servers and initialize
...
+          RuntimePermissionUtils.DROIDBOY_PERMISSION_LOCATION)
+    }
+  }
+}
+```
+
+{% endtab %}
+{% endtabs %}
+
 ### Step 5: Enable Geofences on the Dashboard
```
Comment (davidbielik) [inline _docs/_developer_guide/platform_integration_guides/android/advanced_use_cases/locations_and_geofences.md]: on? or in?

Comment (radixdev) [reply _docs/_developer_guide/platform_integration_guides/android/advanced_use_cases/locations_and_geofences.md]: I think `on`

File: _docs/_developer_guide/platform_integration_guides/android/advanced_use_cases/locations_and_geofences.md (hunk 2 above)
Comment (Bucimis) [reply _docs/_developer_guide/platform_integration_guides/android/advanced_use_cases/locations_and_geofences.md]: Braze Geofence request -> Braze Geofence requests

File: _docs/_developer_guide/platform_integration_guides/android/advanced_use_cases/locations_and_geofences.md [hunk 4]
```diff
@@ -104,12 +104,201 @@ This will cause the SDK to request geofences from Braze's servers and initialize
...
+
+#### Part 2: Manually Request Braze Geofence with GPS Coordinate
+
+Braze Geofences are manually requested via the [`requestGeofences()`][11] method.
+
+{% tabs %}
+{% tab JAVA %}
+
+```java
+Appboy.getInstance(getApplicationContext()).requestGeofences(33.078947d, -116.601356d);
```
Comment (Bucimis) [inline _docs/_developer_guide/platform_integration_guides/android/advanced_use_cases/locations_and_geofences.md]: lol where is this - imo we should use a generic placeholder like
```
Appboy.getInstance(getApplicationContext()).requestGeofences(latitude, longitude);
````
see https://github.com/Appboy/braze-docs/blob/89bfa4b142847929bc1c0c3728ecb1abfededa81/_docs/_developer_guide/platform_integration_guides/android/analytics/setting_custom_attributes.md#setting-a-custom-attribute-with-a-double-value
to match how we format other inputs

Comment (radixdev) [reply _docs/_developer_guide/platform_integration_guides/android/advanced_use_cases/locations_and_geofences.md]: Changed lol

https://www.google.com/maps/search/33.078947,+-116.601356?sa=X&ved=2ahUKEwiz19mZ_b3nAhWxj3IEHbt0AssQ8gEwAHoECAsQAQ

File: _docs/_developer_guide/platform_integration_guides/android/advanced_use_cases/locations_and_geofences.md (hunk 2 above)
Comment (radixdev) [reply _docs/_developer_guide/platform_integration_guides/android/advanced_use_cases/locations_and_geofences.md]: Changed for both

**Pull request merged**